crates/subgraph/src/types/impls.rs

```diff
@@ -1,3 +1,71 @@
+use super::common::*;
+use crate::performance::PerformanceError;
+use alloy::primitives::U256;
+use rain_orderbook_math::BigUintMath;
+use std::str::FromStr;
+
+impl Trade {
+    /// Scales this trade's io to 18 point decimals in U256
+    pub fn scale_18_io(&self) -> Result<(U256, U256), PerformanceError> {
+        let input_amount = if self.input_vault_balance_change.amount.0.starts_with('-') {
+            &self.input_vault_balance_change.amount.0[1..]
+        } else {
+            &self.input_vault_balance_change.amount.0
+        };
+        let output_amount = if self.output_vault_balance_change.amount.0.starts_with('-') {
+            &self.output_vault_balance_change.amount.0[1..]
+        } else {
+            &self.output_vault_balance_change.amount.0
+        };
+        Ok((
+            U256::from_str(input_amount)?.scale_18(
+                self.input_vault_balance_change
+                    .vault
+                    .token
+                    .decimals
```

Comment:
this fallback logic for decimals should be an impl and reused, i can just see someone falling back to `0` somewhere in the app and this falling back to `18` and that being inconsistent across the app, this code duplication on the decimal handling needs to be centralised somewhere in the codebase

Reply:
added